demos/rag_agentic/notebooks/Level2_simple_agentic_with_websearch.ipynb

```diff
@@ -246,15 +574,35 @@
     "            }\n",
     "        ],\n",
     "        session_id=session_id,\n",
+    "        stream=stream_output\n",
     "    )\n",
-    "    for log in EventLogger().log(response):\n",
-    "        log.print()"
+    "    if stream_output:\n",
+    "        for log in EventLogger().log(response):\n",
+    "            log.print()\n",
+    "    else:\n",
+    "        for i, step in enumerate(response.steps):\n",
+    "            step_type = type(step).__name__\n",
+    "            print(\"-\" * 10, f\"📍 Step {i+1}: {step_type}\",\"-\" * 10)\n",
+    "            \n",
+    "            if step_type == \"ToolExecutionStep\":\n",
+    "                print(\"🔧 Executing tool...\")\n",
+    "                pprint(json.loads(step.tool_responses[0].content))\n",
+    "            else:\n",
+    "                if step.api_model_response.content:\n",
+    "                    print(\"🤖 Model Response:\")\n",
+    "                    cprint(f\"{step.api_model_response.content}\\n\", \"yellow\")\n",
+    "                elif step.api_model_response.tool_calls:\n",
+    "                    tool_call = step.api_model_response.tool_calls[0]\n",
+    "                    print(\"🛠️ Tool call Generated:\")\n",
+    "                    cprint(f\"Tool call: {tool_call.tool_name}, Arguments: {json.loads(tool_call.arguments_json)}\", \"yellow\")\n",
+    "        print(\"=\"*10, \"Query processing completed\",\"=\"*10,\"\\n\")"
```

Comment:
Can you turn this whole block into function, like `step_printer(steps)` and add it a new file `src/utils.py` That way we can use it for all the notebooks?